src/dune_rules/opam_create.ml

```diff
@@ -248,14 +248,39 @@ let generate project pkg ~template =
     | None -> ""
     | Some (_, s) -> s )
 
+let all_endings_are_lf_only s =
+  let rec scan last_was_lf i =
+    let c = s.[i] in
+    if last_was_lf && c = '\r' then
+      false
+    else if i = 0 then
+      true
+    else
+      scan (c = '\n') (pred i)
+  in
+  let len = String.length s in
+  len <> 0 && scan true (pred len)
+
+let opam_existing_file_has_lf_endings ~opam_source_path =
+  let open Build.O in
+  Build.if_file_exists opam_source_path
+    ~then_:
+      (let+ contents = Build.contents opam_source_path in
+       all_endings_are_lf_only contents)
+    ~else_:(Build.return false)
+
 let add_rule sctx ~project ~pkg =
   let open Build.O in
   let build_dir = (Super_context.context sctx).build_dir in
-  let opam_path = Path.Build.append_source build_dir (Package.opam_file pkg) in
+  let opam_file = Package.opam_file pkg in
+  let opam_build_path = Path.Build.append_source build_dir opam_file in
+  let opam_source_path = Path.append_source Path.root opam_file in
+  let opam_path = Path.build opam_build_path in
   let opam_rule =
-    (let+ template = opam_template ~opam_path:(Path.build opam_path) in
+    let binary = opam_existing_file_has_lf_endings ~opam_source_path in
```

Comment:
What is the rationale behind this?

It's making the outcome `dune build` stateful, which doesn't seem right to me.